packages/react-composites/src/composites/CallComposite/components/SidePane/SidePaneProvider.tsx

```diff
@@ -0,0 +1,136 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import React, { createContext, useCallback, useContext, useEffect, useMemo } from 'react';
+
+/** @private */
+export type InjectedSidePaneProps =
+  | undefined
+  | {
+      headerRenderer?: () => JSX.Element;
+      contentRenderer?: () => JSX.Element;
+      sidePaneId: string;
+      // Useful to ensure the side pane renders the content of the override even if the side pane is closed.
+      // This avoids remounting the content when the side pane is opened again.
+      hidden?: boolean;
+    };
+
+interface SidePaneProps {
+  /** Side pane header */
+  setHeaderRenderer: React.Dispatch<React.SetStateAction<(() => JSX.Element) | undefined>>;
+  headerRenderer?: () => JSX.Element;
+  /** Side pane content */
+  setContentRenderer: React.Dispatch<React.SetStateAction<(() => JSX.Element) | undefined>>;
+  contentRenderer?: () => JSX.Element;
+  /** tracking open state of the side pane */
+  setActiveSidePaneId: React.Dispatch<React.SetStateAction<string | undefined>>;
+  activeSidePaneId: string | undefined;
+  /** Support injecting content into the side pane from extrnal sources (e.g. CallWithChatComposite's Chat) */
+  setOverrideSidePane: React.Dispatch<React.SetStateAction<InjectedSidePaneProps>>;
+  overrideSidePane: InjectedSidePaneProps;
+}
+
+const defaultSidePaneProps: SidePaneProps = {
+  setHeaderRenderer: () => () => <></>,
+  setContentRenderer: () => () => <></>,
+  // eslint-disable-next-line @typescript-eslint/no-empty-function
+  setActiveSidePaneId: () => {},
+  activeSidePaneId: undefined,
+  // eslint-disable-next-line @typescript-eslint/no-empty-function
+  setOverrideSidePane: () => {},
+  overrideSidePane: undefined
+};
+
+/**
+ * Context for side pane interaction
+ *
+ * @private
+ */
+export const SidePaneContext = createContext<SidePaneProps>(defaultSidePaneProps);
+
+/**
+ * Props to LocalizationProvider
+ *
+ * @private
+ */
+export type SidePaneProviderProps = {
+  children: React.ReactNode;
+};
+
+/**
+ * Provider to provide localized strings for this library's composites.
+ *
+ * @private
+ */
+export const SidePaneProvider = (props: SidePaneProviderProps): JSX.Element => {
+  const [headerRenderer, setHeaderRenderer] = React.useState<(() => JSX.Element) | undefined>();
+  const [contentRenderer, setContentRenderer] = React.useState<(() => JSX.Element) | undefined>();
+  const [activeSidePaneId, setActiveSidePaneId] = React.useState<string>();
```

Comment:
Discussion from last time - we could still use provider, but just not to add state here
So this provider could take props(either states or normal props) from outside of the composite, then store it here

Or we just remove provider then use a top to down state from composite to each single component

Reply:
Updated for provider to take props for now